otx/algorithms/classification/tasks/inference.py

```diff
@@ -303,6 +303,9 @@ def _init_recipe(self):
         if train_type == TrainType.INCREMENTAL:
             recipe = os.path.join(recipe_root, "incremental.yaml")
 
+        if self._hyperparams.learning_parameters.enable_supcon:
```

Comment:
If supcon only supports incremental learning(fine-tuning), this blocks should be inside if train_type == TrainType.INCREMENTAL. Because semi-sl and self-sl will be joined

Reply:
Right...